include/seqan3/utility/bloom_filter/bloom_filter.hpp

```diff
@@ -0,0 +1,397 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2020, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2020, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \author Tobias Loka <tobias.loka AT hpi.de>
+ * \brief Provides seqan3::bloom_filter.
+ */
+
+#pragma once
+
+#include <seqan3/search/dream_index/interleaved_bloom_filter.hpp>
```

Comment:
```suggestion
#include <seqan3/search/dream_index/interleaved_bloom_filter.hpp>
```

It would be desirable if we do not depend on search at all.

The question is how much we need to untangle to just include the entities we need.

Entities seen so far:
* seqan3::data_layout
* seqan3::bin_size
* seqan3::hash_function_count
* sdsl::bit_vector

Reply:
The seqan3 entities could be moved into this utility/dream_index folder (one file per entity) making utility independent again.

Reply:
I think for `seqan3::data_layout` and `seqan3::hash_function_count`, this would make sense. For `seqan3::bin_size` on the other hand, I have a bit of the feeling that I am "abusing" this here anyways (as we don't have any bins...).  
I am not sure how important it is for your design to work with `strong_type`, but in principle we could either just use a `size_t` here, or define its own `strong_type`!? I guess, the `strong_type` is more important when having more than one numeric parameter, isn't it?

```
bloom_filter(size_t size, // should this be renamed to bit_size as well?
             seqan3::hash_function_count funs = seqan3::hash_function_count{2u})
```